src/map/options.ts

```diff
@@ -154,12 +157,15 @@ export class MapOptions extends OptionsGroup {
         this._modal = modal;
         this._openModal = openModal;
         root.appendChild(this._openModal);
-        document.body.appendChild(this._modal);
 
         this._bind(properties);
         this.applySettings(settings);
     }
 
+    public getById<T extends HTMLElement = HTMLElement>(id: string): T {
+        return this._modal.shadow.getElementById(id) as T;
+    }
```

Comment:
The previous implementation of `getById` also checked if the returned element is not `null`, which caught a few bugs in, the past. 

It also takes the root in which to search the element as a second argument, so you could implement this function using `getById(this.getId(id), this._modal.shadow)` where needed.

Since the function is only looking inside the modal, how about calling it `getModalElement` or something similar? And we can also merge this function with `getId`.

Reply:
Yes I agree it's better to use the global `getByID` method which I actually had updated to work with shadow roots (although I'm a little skeptical to the idea of using the second argument when not looking inside a shadow root, given that ids are unique).

I also ended up removing all guids in this file so there is no more need for `getId()`.

Reply:
> although I'm a little skeptical to the idea of using the second argument when not looking inside a shadow root, given that ids are unique

The use case for this is that the element might not be inserted in the document yet, which is the case when creating a new jupyter widget

Reply:
Ok this makes sense indeed!